src/v/cluster/tm_stm.cc

```diff
@@ -167,11 +167,66 @@ ss::future<checked<model::term_id, tm_stm::op_status>> tm_stm::do_barrier() {
         });
 }
 
-ss::future<> tm_stm::checkpoint_ongoing_txs() { return ss::now(); }
+ss::future<> tm_stm::checkpoint_ongoing_txs() {
+    if (!use_new_tx_version()) {
+        co_return;
+    }
+    auto cmp = [](const auto& t1, const auto& t2) { return t1.id < t2.id; };
+    std::set<tm_transaction, decltype(cmp)> all_txes;
+
+    auto can_transfer = [](const tm_transaction& tx) {
+        return !tx.transferring
+               && (tx.status == tm_transaction::ready || tx.status == tm_transaction::ongoing);
+    };
+
+    // Loop through all ongoing/pending txns in memory and checkpoint.
+
+    for (auto& [_, tx] : _log_txes) {
```

Comment:
btw, can this loops stall reactor time for long time?

Reply:
think not.. a typical working set of the txns is quite small (see the latest rev where we only need to worry about _mem_txes which is even smaller). Also the loop body doesn't do anything costly.